client/src/components/App.jsx

```diff
@@ -26,8 +29,8 @@ class App extends React.Component {
   }
 
   render() {
-    const { products } = this.state;
-    return <ProductDetails products={products} />;
+    const { product } = this.state;
```

Comment:
This is a preference but it seems redundant to create a variable called { product } if you are only going to use it once. I'd recommend just writing the following:
return <ProductDetails product={this.state.product} />;